### DISCUSSION TITLE
Is a feature executed before or after the actual Dockerfile?

### DISCUSSION THREAD

**Discussion by 8192K** [devcontainers/community#121] [Q&A]
It seems that the basic feature I added to my dev container setup (common-utils) is only executed after the actual Dockerfile. This means I cannot use the non-root user the feature creates in my Dockerfile.

Is this a correct assumption? How can I change the execution order so the feature runs before the actual Dockerfile?

**Comment by samruddhikhandale** [comment]
Hi 👋 

> This means I cannot use the non-root user the feature creates in my Dockerfile.

Yes, that's expected as Features are always executed after the Dockerfile is executed.

> How can I change the execution order so the feature runs before the actual Dockerfile?

Unfortunately, there is no current way to reverse the flow. However, how about adding the logic from your Dockerfiles to a [lifecycle hook](https://containers.dev/implementors/json_reference/#lifecycle-scripts) or to a [local Feature](https://containers.dev/implementors/features-distribution/#addendum-locally-referenced)?

**Reply by 8192K** [reply]
Thank you. I could not find any mention of the execution order in the docs. Could you point me to it please?
